os/board/common/partitions.c

```diff
@@ -121,17 +123,25 @@ void configure_partitions(void)
 		}
 
 #if defined(CONFIG_MTD_PARTITION_NAMES)
-		if (*names) {
-			mtd_setpartitionname(mtd_part, names);
-		}
 
-		while (*names) {
-			if (*names == ',') {
+		if (*names) {
```

Comment:
`while(*name)` can check that *name is NULL or not. 
So I think this `if` is not needed.